### PR TITLE
[Order Details] Reuse payment methods from Order Details

### LINKED ISSUE #6782: [Mobile Payments] Make Payment Methods screen reusable outside the Simple Payments flow
See pdfdoF-Hq-p2 for context and discussion.

Simple Payments Payment Method Selection

This screen will be reused from Order Details and could be used anywhere else that merchants need to accept payments, not just the Simple Payments flow.

This will make Cash and Payment Link more accessible in other parts of the app, and mean we can always allow merchants to take payment in some way, even if there is an issue with Card Present Payments.

- [ ] Ensure Analytics are logged correctly relating to the payment flow in use (i.e. from Simple Payments vs from Order Details. Possibly also Order Creation, though that may not need to be differentiated from Order Details.)

### PR DESCRIPTION
**Pull request by joshheald**
Closes: #6782 
Part of: #7003 

### Description

This change renames and reuses the Simple Payments Payment Method screen from Order Details.

Merchants will now see a `Collect Payment` button for any unpaid order even if they aren't set up for Card-Present Payments. Currently, they still need to be Card-present eligible to see the `Collect Payment` button, but that requirement will be removed in a future PR.

Tapping Collect Payment shows the Payment Method screen and allows them to choose between Cash, Payment Link, and (if available) Card payment.

### Testing instructions

Using a US store with WCPay or Stripe

1. Go to the Orders tab, and add a new order using the + button
2. Use `Create Order` and set up the order
3. At the end of the flow, you'll be on the Order Details screen and should see `Collect Payment`
4. Tap `Collect Payment` and observe that you are shown the payment selection screen
5. Tap `Card` and take a card payment.
6. Observe that you're returned to the Order Details screen with the order paid when this is complete

N.B there is an outstanding issue where the `Collect Payment` button remains visible after the payment flow is complete. This will be addressed in the next PR.

Repeat the above using `Card` and `Payment Link`. Note that `Payment Link` does not mark the order as paid, as it requires further action by the customer to follow the link and make payment.

### Screenshots

https://user-images.githubusercontent.com/2472348/174629140-7108171b-1158-4a34-949a-d5e8aecb61ef.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift [hunk 1]
```diff
@@ -692,36 +692,14 @@ extension OrderDetailsViewModel {
...
-                    self?.collectPaymentsUseCase = nil
-                })
-        }
+        let paymentMethodsViewModel = PaymentMethodsViewModel(siteID: order.siteID,
+                                                              orderID: order.orderID,
+                                                              paymentLink: order.paymentURL,
+                                                              formattedTotal: order.total,
+                                                              flow: .orderPayment)
+        let paymentMethodsViewController = PaymentMethodsHostingController(viewModel: paymentMethodsViewModel)
+        let paymentMethodsNC = WooNavigationController(rootViewController: paymentMethodsViewController)
```
Comment (toupper) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift:701]: Not a blocker of course, but perhaps we could avoid the NC acronym here? It took me a few seconds to find out what it meant with that: `paymentMethodsNC` -> `paymentMethodsNavigationController`

Comment (toupper) [inline WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift]: Not a blocker of course, but perhaps we could avoid the NC acronym here? It took me a few seconds to find out what it meant with that: `paymentMethodsNC` -> `paymentMethodsNavigationController`

File: WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift [hunk 2]
```diff
@@ -1,10 +1,6 @@
-import Foundation
 import SwiftUI
 
-/// View to choose what payment method will be used with the simple payments order.
-///
-struct SimplePaymentsMethod: View {
-
+struct PaymentMethodsView: View {
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsView.swift:3]: Nice rename here to make it more generic 👍

File: WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsViewModel.swift [hunk 3]
```diff
@@ -252,10 +258,24 @@ final class SimplePaymentsMethodsViewModel: ObservableObject {
         presentNoticeSubject.send(.created)
         trackFlowCompleted(method: .paymentLink)
     }
+
+    /// Track the flow cancel scenario.
+    ///
+    func userDidCancelFlow() {
+        analytics.track(event: WooAnalyticsEvent.PaymentsFlow.paymentsFlowCanceled(flow: flow))
+    }
+
```
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Orders/Payment Methods/PaymentMethodsViewModel.swift:267]: Few empty lines here? :)

**Comment by toupper** [comment]
LGTM and tests well, with Cash, Card, and Share link 🥳 Just added a few comments in code, nothing big.
I am wondering if you consider creating a feature flag for this, so we don't end up creating a last big pull request to `trunk` and review the code twice :)

**Review by toupper — APPROVED** [review]

**Comment by joshheald** [comment]
> I am wondering if you consider creating a feature flag for this, so we don't end up creating a last big pull request to `trunk` and review the code twice :)

Thanks @toupper. I wasn't intending on making a feature flag, nor getting things double-reviewed... each branch is based on the previous PR, so once #7103 is reviewed & merged, the rest will all merge down to trunk cleanly. I will do a merge from trunk first though, as it's a while since they were branched off trunk.

**Pull request merged**